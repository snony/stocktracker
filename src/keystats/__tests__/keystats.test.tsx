import { mockFailedGlobalState, mockGlobalState } from '__mock__/globalstate.mock'
import { shallow } from 'enzyme'
import toJSON from 'enzyme-to-json'
import KeyStats from 'keystats/keystats'
import React from 'react'

describe('Keystats Component', () => {
  it('should render key stats when keystats state updated', () => {
    const keystats = mockGlobalState.keystats
    const wrapper = shallow(<KeyStats keystats={keystats} />)

    expect(toJSON(wrapper)).toMatchSnapshot()
  })

  it('should render status when keystats state is not SUCCESS', () => {
    const keystats = mockFailedGlobalState.keystats
    const wrapper = shallow(<KeyStats keystats={keystats} />)

    expect(toJSON(wrapper)).toMatchSnapshot()
  })
})
