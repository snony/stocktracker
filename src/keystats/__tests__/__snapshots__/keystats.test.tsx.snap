// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Keystats Component should render key stats when keystats state updated 1`] = `
<div
  className="keystats-container"
>
  <div
    className="keystats-container__stat"
    key="Previous Close"
  >
    <span
      className="label label--small"
    >
      Previous Close
    </span>
    <span
      className="label"
    >
      10
    </span>
  </div>
  <div
    className="keystats-container__stat"
    key="Day Range"
  >
    <span
      className="label label--small"
    >
      Day Range
    </span>
    <span
      className="label"
    >
      10 - 100
    </span>
  </div>
  <div
    className="keystats-container__stat"
    key="Volume"
  >
    <span
      className="label label--small"
    >
      Volume
    </span>
    <span
      className="label"
    >
      10,000
    </span>
  </div>
  <div
    className="keystats-container__stat"
    key="Market Cap"
  >
    <span
      className="label label--small"
    >
      Market Cap
    </span>
    <span
      className="label"
    >
      100,000,000
    </span>
  </div>
  <div
    className="keystats-container__stat"
    key="P/E Ratio"
  >
    <span
      className="label label--small"
    >
      P/E Ratio
    </span>
    <span
      className="label"
    >
      10
    </span>
  </div>
  <div
    className="keystats-container__stat"
    key="Open"
  >
    <span
      className="label label--small"
    >
      Open
    </span>
    <span
      className="label"
    >
      10
    </span>
  </div>
  <div
    className="keystats-container__stat"
    key="52 Week Range"
  >
    <span
      className="label label--small"
    >
      52 Week Range
    </span>
    <span
      className="label"
    >
      10 - 100
    </span>
  </div>
  <div
    className="keystats-container__stat"
    key="Total Avg. Volume"
  >
    <span
      className="label label--small"
    >
      Total Avg. Volume
    </span>
    <span
      className="label"
    >
      10,000
    </span>
  </div>
  <div
    className="keystats-container__stat"
    key="Earnings Per Share"
  >
    <span
      className="label label--small"
    >
      Earnings Per Share
    </span>
    <span
      className="label"
    >
      10
    </span>
  </div>
  <div
    className="keystats-container__stat"
    key="Dividend & Yield"
  >
    <span
      className="label label--small"
    >
      Dividend & Yield
    </span>
    <span
      className="label"
    >
      10%
    </span>
  </div>
</div>
`;

exports[`Keystats Component should render status when keystats state is not SUCCESS 1`] = `
<div
  className="keystats-container"
>
  <span
    className="label"
  >
    FAILED
  </span>
</div>
`;
